jsx/FilterForm.js

```diff
@@ -0,0 +1,203 @@
+          }
+        }
+        // Pass onUserInput and value props to all children
+        formElements.push(React.cloneElement(child, {
+          onUserInput: callbackFunc,
+          value: filterValue ? filterValue : '',
+          key: key
+        }));
+        // Initialize filter for StaticDataTable
+        this.setTableFilter(elementName, child.ref, filterValue);
+      } else {
+        formElements.push(React.cloneElement(child, {key: key}));
+      }
+    }.bind(this));
+
+    return formElements;
+  }
+
+  /**
+   * Appends entry to tableFilter object or deletes it if value is
+   * empty.
+   *
+   * Sets exactMatch to true for all SelectElements (i.e dropdowns)
+   * in order to force StaticDataTable to do exact comparaison
+   *
+   * @param {string} type - form element type (i.e component name)
+   * @param {string} key - the name of the form element
+   * @param {string} value - the value of the form element
+   *
+   * @return {{}} tableFilter - filterData
+   */
+  setTableFilter(type, key, value) {
+    // Deep copy of tableFilter object
+    let tableFilter = JSON.parse(JSON.stringify(this.tableFilter));
+
+    if (key && value) {
+      tableFilter[key] = {};
+      tableFilter[key].value = value;
+      tableFilter[key].exactMatch = (type === "SelectElement");
+    } else if (key && value === '') {
+      delete tableFilter[key];
+    }
+
+    // Update class variable
+    this.tableFilter = tableFilter;
+
+    return tableFilter;
+  }
+
+  /**
+   * Sets filter object and querystring to reflect values of input fields
+   *
+   * @param {string} type - form element type (i.e component name)
+   * @param {string} fieldName - the name of the form element
+   * @param {string} fieldValue - the value of the form element
+   */
+  setFilter(type, fieldName, fieldValue) {
+    // Make sure both key/value are string before sending them to querystring
+    if (typeof fieldName !== "string" || typeof fieldValue !== "string") {
+      return;
+    }
+
+    // Update query string and get new filter object
+    let filter = QueryString.set(this.state.filter, fieldName, fieldValue);
+
+    // Update tableFilter and get new tableFilter object
+    let tableFilter = this.setTableFilter(type, fieldName, fieldValue);
+
+    this.props.onUpdate(tableFilter);
+    this.setState({filter});
+  }
+
+  toggleCollapsed() {
+    this.setState({collapsed: !this.state.collapsed});
+  }
+
+  render() {
+    // Selection filter open by default
+    let glyphClass = "glyphicon pull-right glyphicon-chevron-up";
+    let panelClass = "panel-collapse collapse in";
+
+    // Change arrow direction when closed
+    if (this.state.collapsed) {
+      glyphClass = "glyphicon pull-right glyphicon-chevron-down";
+    }
+
+    // Get formatted children
+    let formElements = this.getFormElements();
+
+    return (
+      <div className="panel panel-primary">
+        <div className="panel-heading"
```

Comment:
Note: This ideally should be replaced by [React Panel](161229_react_panel), once it is merged into the main codebase